Tasks/SonarQubePostTest/PRCA/ReportProcessor.ps1

```diff
@@ -0,0 +1,212 @@
+
+    if ($tokens.Count -ne 4) 
+    {
+        throw "Internal error: component $component is not in the expected format (expected 4 parts)"
+    }
+
+    #third token must be a guid
+    $guidToken = $tokens[2]
+
+    $outGuid = New-Object -TypeName "System.Guid"
+    if (![System.Guid]::TryParse($guidToken, [ref]$outGuid))
+    {
+        throw "Internal error: $guidToken is not a GUID"
+    }
+
+    return $guidToken
+}
+
+#
+# Returns a path relative to the repo root for a file which has new code analysis issue(s)
+#
+function GetPathRelativeToRepoRoot($component)
+{
+    if ($component -and $ComponentKeyAndRelativePathCache.ContainsKey($component))
+    {
+        $relativeFilePath = $ComponentKeyAndRelativePathCache[$component]
+        Write-Verbose "GetPathRelativeToRepoRoot: Found cached entry, returning data from cache, relativePath:$relativeFilePath"
+
+        return $relativeFilePath
+    }
+
+    #sonar runner creates the component value as '[SonarQube project key]:[SonarQube project value]:[MSBuild project guid]:[file name relative to MSBuild project file path]'
+   
+    #third token must be guid
+    $guidToken = $tokens[2]
+    Write-Verbose "GetPathRelativeToRepoRoot: guidToken:$guidToken"
+
+    if (!$ProjectGuidAndFilePathMap.ContainsKey($guidToken))
+    {
+        Write-Verbose "GetPathRelativeToRepoRoot: An entry for project guid $guidToken could not be found, check ProjectInfo.xml file"
+        return $null
+    }
+    if (!$ComponentKeyAndPathMap.ContainsKey($component))
+    {
+        Write-Verbose "GetPathRelativeToRepoRoot: An entry for component key $component could not be found, check sonar-report.json file"
+        return $null
+    }
+
+    #This stores the full on-disk path of the *.xxproj file
+    $projectPath = $($ProjectGuidAndFilePathMap[$guidToken])
+
+    $finalFilePath = [System.IO.Path]::GetDirectoryName($projectPath)
+    
+    $finalFilePath = [System.IO.Path]::Combine($finalFilePath, $ComponentKeyAndPathMap[$component])
+    Write-Verbose "GetPathRelativeToRepoRoot: finalFilePath:$finalFilePath"
+
+    $repoLocalPath = GetTaskContextVariable "Build.Repository.LocalPath"
+    
+    if (!$repoLocalPath)
+    {
+        throw "GetPathRelativeToRepoRoot: Could not get task variable Build.Repository.LocalPath"
+    }
+
+    Write-Verbose "GetPathRelativeToRepoRoot: repoLocalPath:$repoLocalPath"
+
+    #this will remove from the file path, the part upto the repo name. 
+    #e.g. finalFilePath=C:\Agent\_work\ef030e14\s\Mail2Bug\Main.cs and repoLocalPath=C:\Agent\_work\ef030e14\s
+    #after the SubString() call finalFilePath=\Mail2Bug\Main.cs
+    $finalFilePath = $finalFilePath.ToString().SubString($repoLocalPath.Length);
+
+    #Replace '\' with '/'. VSO expects file path like /Mail2Bug/Main.cs (\Mail2Bug\Main.cs does not work)
+    $finalFilePath = $finalFilePath.ToString().Replace([System.IO.Path]::DirectorySeparatorChar, [System.IO.Path]::AltDirectorySeparatorChar)
```

Comment:
Same as above + i think that uri.makerelative actually returns in the format that you need (with alt dir separator)